### PR TITLE
Elastic 5.5 support

### PR DESCRIPTION
**Pull request by nomoa**
- merge master
- add support for 5.5 (InternalSearchHit -> SearchHit)
- remove usage of deprecated native scripts

### CODE REVIEW

**Comment by nomoa** [comment]
@worleydl @gehel mind having a quick look? Thanks!

File: src/main/java/com/o19s/es/ltr/query/StoredLtrQueryBuilder.java [hunk 1]
```diff
@@ -50,6 +50,7 @@
     public static final ParseField STORE_NAME = new ParseField("store");
     public static final ParseField PARAMS = new ParseField("params");
 
+    private final transient FeatureStoreLoader storeLoder;
```
Comment (worleydl) [inline src/main/java/com/o19s/es/ltr/query/StoredLtrQueryBuilder.java]: Should this be storeLoader?

File: src/main/java/com/o19s/es/ltr/query/StoredLtrQueryBuilder.java [hunk 2]
```diff
@@ -84,11 +93,12 @@ protected void doWriteTo(StreamOutput out) throws IOException {
...
 
-    public static Optional<StoredLtrQueryBuilder> fromXContent(QueryParseContext context) throws IOException {
+    public static Optional<StoredLtrQueryBuilder> fromXContent(FeatureStoreLoader storedLoader,
+                                                               QueryParseContext context) throws IOException {
         XContentParser parser = context.parser();
-        final StoredLtrQueryBuilder builder;
+        final StoredLtrQueryBuilder builder =  new StoredLtrQueryBuilder(storedLoader);
         try {
-            builder = PARSER.parse(context.parser(), null);
+            PARSER.parse(context.parser(), builder,null);
```
Comment (worleydl) [inline src/main/java/com/o19s/es/ltr/query/StoredLtrQueryBuilder.java]: Whitespace nitpick

File: src/main/java/com/o19s/es/ltr/query/StoredLtrQueryBuilder.java (hunk 1 above)
Comment (nomoa) [inline src/main/java/com/o19s/es/ltr/query/StoredLtrQueryBuilder.java]: doh... you're right fixing.

File: src/main/java/com/o19s/es/ltr/query/StoredLtrQueryBuilder.java (hunk 2 above)
Comment (nomoa) [inline src/main/java/com/o19s/es/ltr/query/StoredLtrQueryBuilder.java]: done

File: src/main/java/com/o19s/es/ltr/query/StoredLtrQueryBuilder.java [hunk 3]
```diff
@@ -67,9 +68,17 @@
         declareStandardFields(PARSER);
     }
 
-    public StoredLtrQueryBuilder() {}
-    public StoredLtrQueryBuilder(StreamInput input) throws IOException {
+    public StoredLtrQueryBuilder() {
```
Comment (gehel) [inline src/main/java/com/o19s/es/ltr/query/StoredLtrQueryBuilder.java]: I assume that the default empty constructor is needed for some reason, but can't actually be used? And it must not throw an exception at construction time? A comment explaining why we allow building what looks like an unusable StoredLtrQueryBuilder would be welcomed.

Comment (nomoa) [inline src/main/java/com/o19s/es/ltr/query/StoredLtrQueryBuilder.java]: fair point

Comment (gehel) [inline src/main/java/com/o19s/es/ltr/query/StoredLtrQueryBuilder.java]: Is it really only for testing? If that's the case, wouldn't it make more sense to inject this fake store loader in the test? Or at least tag this constructor with `@VisibleForTesting`?

Comment (nomoa) [inline src/main/java/com/o19s/es/ltr/query/StoredLtrQueryBuilder.java]: I see what you mean, I'll move this behaviors to the test, it'll me give a chance to throw some hints on how to use it in integration tests.

**Pull request merged**